system/session/drivers/cookie.php

```diff
@@ -47,7 +48,7 @@ public function write($id, $cargo) {
 		$serialized = serialize($cargo);
 
 		// create a signature to verify content when unpacking
-		$sign = hash('md5', $serialized);
+		$sign = hash_hmac('sha256', $serialized, $session_key, true);
```

Comment:
It would log everyone out, which is a minor one time inconvenience in
exchange for much better security
On Jul 30, 2015 7:17 PM, "Jarod Brennfleck" <notifications@github.com>
wrote:

> In system/session/drivers/cookie.php
> <https://github.com/anchorcms/anchor-cms/pull/904#discussion_r35932548>:
>
> > @@ -47,7 +48,7 @@ public function write($id, $cargo) {
> >  		$serialized = serialize($cargo);
> >
> >  		// create a signature to verify content when unpacking
> > -		$sign = hash('md5', $serialized);
> > +		$sign = hash_hmac('sha256', $serialized, $session_key, true);
>
> Even though it is stronger, wouldn't this cause conflicts because we're
> starting to use a different algorithm?
> (Pseudo)
> (hash('md5', 'qweasd') === hash('sha256', 'qweasd')) = False
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anchorcms/anchor-cms/pull/904/files#r35932548>.
>